### ISSUE TITLE
How can I get attributes when using Value-Object?

### ISSUE THREAD

**Issue by RomainLanz** [fruux/sabre-xml#104]
Dear !

I'm trying to retrieve the attributes of my element when using Value-Object serializer.
How this can be done?

--

**Example:**
```xml
<competition code="XXXX">
  <yyy></yyy>
</competition>
```

```php
$service = new Service();
$service->mapValueObject('{}Competition', Competition::class);
$parsedXml = $service->parse($xml)

$parsedXml->code // Undefined
$parsedXml->_code // Undefined
$parsedXml->attributes // Undefined
```

Thanks!

**Comment by staabm** [comment]
you can't.

**Comment by RomainLanz** [comment]
Thank you @staabm.

This should be implemented in my opinion.

**Comment by staabm** [comment]
@RomainLanz I agree with you. 

@evert and I already discussed this somewhere but I cannot find the issue right now.

**Comment by evert** [comment]
The reason we're not doing this is that the value-object mapping depends on being able to map from xml to a PHP object and back. If attributes are in the mix, we no longer have a way to to know whether a php object property should be encoded  as an attribute, or as a sub-element.

So what would it look like if we had 'support' for this? Well, I think we would end up with the API we already have. We have support for complicated XML structures, because we provide all the functions that allow you to encode this. It's not as simple as the value-object mapping simply because we cannot predict your intention.

So we support this just fine, you just need to write your own serializer, which should be pretty easy. If it's not easy though, let me know... I'm interested in improving the api if we can find a good way to do so =)

**Comment by RomainLanz** [comment]
Finally I have used [SimpleXML](http://php.net/manual/en/book.simplexml.php) base class to do the job.

I can easily go through my XML with class oriented syntax and retrieve attributes.

**Comment by evert** [comment]
That's too bad, I think you misunderstood the earlier comments. It's entirely possible to get access to attributes, just not with the value-object mapper, which is really just intended for the simplest cases.

Anyway, glad you figured it out!

**Comment by RomainLanz** [comment]
I completely understood your previous comment.

I have used the default reader to parse my XML but it was a pain in the ass to retrieve every information of it (array is not that good).

Maybe I have misused the package and I could have done my job by using it differently... the documentation didn't show me another and I didn't have the time to configure or search.

Anyway [SimpleXML](https://secure.php.net/manual/en/book.simplexml.php) does perfectly the job and everything works fine now!
